src/pages/en/guides/content-collections.mdx

```diff
@@ -396,6 +396,127 @@ const { Content } = await entry.render();
+If your existing site uses [file-based routing](/en/core-concepts/astro-pages/#markdownmdx-pages) with Markdown files in the `src/pages` folder, you may want to migrate to content collections. This gives you the benefit of type-safe frontmatter, but requires you to set up routing yourself. This guide uses the result of the [Build a Blog tutorial](/en/tutorial/0-introduction/) as an example starting point.
+
+1. [Upgrade](/en/guides/upgrade-to/v2/) to Astro v2.0 or later, and upgrade all integrations to their latest versions. 
+
+2. [Set up TypeScript](/en/guides/content-collections/#setting-up-typescript) for content collections.
+3. Move Markdown and MDX pages from `src/pages` to subdirectories of `src/content`. Each subdirectory represents a content type. For example, to migrate the [blog posts in the tutorial](/en/tutorial/2-pages/2/), move the contents of `src/pages/posts` to `src/content/posts`. 
+4. Create a `src/content/config.ts` file and [define a schema](/en/guides/content-collections/#defining-a-collection-schema) for each content type. For the blog, we only have one content type, `posts`:
+
+```ts title="src/content/config.ts"
+// Import utilities from `astro:content`
+import { z, defineCollection } from "astro:content";
+// Define a schema for each collection you'd like to validate.
+const postsCollection = defineCollection({
+    schema: z.object({
+      title: z.string(),
+      pubDate: z.date(),
+      description: z.string(),
+      author: z.string(),
+      image: z.object({
+        url: z.string(),
+        alt: z.string()
+      }),
+      tags: z.array(z.string())
+    })
+ });
+// Export a single `collections` object to register your collection(s)
+export const collections = {
+  posts: postsCollection,
+};
+```
+
+:::tip
+If your editor doesn't recognize `astro:content`, make sure you're on the latest version of Astro and try restarting the dev server. 
+:::
+
+5. To [generate routes](/en/guides/content-collections/#generating-routes-from-content), create a `.astro` page with a [dynamic route](/en/core-concepts/routing/#dynamic-routes) in `src/pages/posts` to [query and render your content](#querying-collections). Our tutorial blog uses the static output mode, so we'll use `getStaticPaths()` to create a route for each post:
+
+```astro title="src/pages/posts/[...slug].astro"
+---
+import { getCollection } from 'astro:content';
+import MarkdownPostLayout from '../../layouts/MarkdownPostLayout.astro';
+
+export async function getStaticPaths() {
+  const blogEntries = await getCollection('posts');
+  return blogEntries.map(entry => ({
+    params: { slug: entry.slug }, props: { entry },
+  }));
+}
+
+const { entry } = Astro.props;
+const { Content } = await entry.render();
+---
+<MarkdownPostLayout frontmatter={entry.data}>
+  <Content />
+</MarkdownPostLayout>
+```
+
+6. Since we wrapped the content in a layout in the dynamic route, remove the `layout` definition in each post's frontmatter.
```

Comment:
```suggestion
6. Remove the `layout` definition in each individual post's frontmatter. This `layout` property is only available to Markdown and MDX files within `src/pages/`. Instead, a layout is specified for all of your generated routes at once.
```

Reply:
It isn't true that the `layout` property functionality isn't _available_ with content collections. It will still work: when you call `entry.render()`, if the entry has a layout property, it will render it with a layout.

So it's not that you _can't_ do it, but since all of the posts specify the same layout, we might as well "pull it out" and wrap it ourselves.

Reply:
Made an edit that hopefully reflects your comments!